src/wp-includes/option.php

```diff
@@ -783,15 +783,42 @@ function update_option( $option, $value, $autoload = null ) {
 	$value = apply_filters( 'pre_update_option', $value, $option, $old_value );
 
 	/*
-	 * If the new and old values are the same, no need to update.
-	 *
 	 * Unserialized values will be adequate in most cases. If the unserialized
 	 * data differs, the (maybe) serialized data is checked to avoid
 	 * unnecessary database calls for otherwise identical object instances.
 	 *
 	 * See https://core.trac.wordpress.org/ticket/38903
 	 */
-	if ( $value === $old_value || maybe_serialize( $value ) === maybe_serialize( $old_value ) ) {
+
+	if ( maybe_serialize( $value ) === maybe_serialize( $old_value ) ) {
+		return false;
+	}
```

Comment:
Should we move this statement below the new `foreach` loop? I believe the `maybe_serialize()` calls are more expensive, and originally they were also running after the `$value === $old_value` logic, which is what the new `foreach` replaces.

Reply:
Thanks for feedback. Addressed in https://github.com/WordPress/wordpress-develop/pull/5139/commits/00a66a7ec284abd70e9e5d337f34121bce93c61b